src/test/java/ch/wisv/areafiftylan/products/service/TicketServiceTest.java

```diff
@@ -0,0 +1,664 @@
+    }
+
+    @Test
+    public void requestTicketOfTypeObjectsOptionsNull() {
+        User user = persistUser();
+        TicketType type = ticketTypeRepository.findByName(TEST_TICKET).get();
+        Ticket ticketToPersist = new Ticket(user, type);
+        List<TicketOption> options = null;
+        Set<TicketOption> optionSet = new HashSet<>();
+        testEntityManager.persist(ticketToPersist);
+
+        Ticket ticket = ticketService.requestTicketOfType(user, type, options);
+
+        assertEquals(user, ticket.getOwner());
+        assertEquals(optionSet, ticket.getEnabledOptions());
+        assertEquals(type, ticket.getType());
+    }
+
+    @Test
+    public void setupForTransfer() {
+        User goalUser = persistUser();
+        String goalUsername = goalUser.getUsername();
+        Ticket ticket = persistTicket();
+        Long ticketId = ticket.getId();
+
+        TicketTransferToken ttt = ticketService.setupForTransfer(ticketId, goalUsername);
+
+        assertEquals(ticket, ttt.getTicket());
+        assertEquals(goalUser, ttt.getUser());
+        assertEquals(true, ttt.isValid());
+    }
+
+    @Test
+    public void setupForTransferDuplicate() {
+        User goalUser = persistUser();
+        String goalUsername = goalUser.getUsername();
+        Ticket ticket = persistTicket();
+        Long ticketId = ticket.getId();
+
+        thrown.expect(DuplicateTicketTransferTokenException.class);
+        thrown.expectMessage(" is already set up for transfer!");
+
+        ticketService.setupForTransfer(ticketId, goalUsername);
+        ticketService.setupForTransfer(ticketId, goalUsername);
+    }
+
+    @Test
+    public void setupForTransferAlreadyLinked() {
+        User goalUser = persistUser();
+        String goalUsername = goalUser.getUsername();
+        Ticket ticket = persistTicket();
+        Long ticketId = ticket.getId();
+        testEntityManager.persist(new RFIDLink("1234567890", ticket));
+
+        thrown.expect(TicketAlreadyLinkedException.class);
+        thrown.expectMessage("Ticket has already been linked to a RFID");
+
+        ticketService.setupForTransfer(ticketId, goalUsername);
+    }
+
+    @Test
+    public void transferTicket() {
+        User goalUser = persistUser();
+        String goalUsername = goalUser.getUsername();
+        Ticket ticket = persistTicket();
+        Long ticketId = ticket.getId();
+        TicketTransferToken ttt = ticketService.setupForTransfer(ticketId, goalUsername);
+        String token = ttt.getToken();
+
+        ticketService.transferTicket(token);
+
+        assertEquals(goalUser, ticket.getOwner());
+        assertFalse(ttt.isUnused());
+    }
+
+    @Test
+    public void transferTicketTokenNotFound() {
+        thrown.expect(TokenNotFoundException.class);
+        thrown.expectMessage("Could not find token ");
+
+        ticketService.transferTicket("invalid_token");
+    }
+
+    @Test
+    public void transferTicketTokenNull() {
```

Comment:
It is. However, in OrderServiceTest, testing of invalid and null is also done, and these tests are heavily inspired by it.